lib/internal/modules/run_main.js

```diff
@@ -81,6 +81,14 @@ function shouldUseESMLoader(mainPath) {
   if (mainPath && StringPrototypeEndsWith(mainPath, '.mjs')) { return true; }
   if (!mainPath || StringPrototypeEndsWith(mainPath, '.cjs')) { return false; }
 
+  if (getOptionValue('--experimental-strip-types')) {
+    // This ensures that --experimental-default-type=commonjs and .mts files are treated as commonjs
+    if (getOptionValue('--experimental-default-type') === 'commonjs') { return false; }
+    if (mainPath && StringPrototypeEndsWith(mainPath, '.cts')) { return false; }
+    // This will likely change in the future to start with commonjs loader by default
```

Comment:
Why use the CommonJS loader at all? Why not have any of the three TypeScript extensions cause the ESM loader to handle the main entry point, and then that’s the only place we need to implement type stripping support?

Yes the CommonJS loader is currently slightly faster, but it doesn’t need to be so; there’s room for improving the ESM loader to reach parity or near-parity. But more to the point, the vast majority of TypeScript code is written using ESM syntax, so I don’t think we need to optimize for the case of someone writing TypeScript with `require` and `module.exports`.

Reply:
I guess since we support both loaders, the right thing is to support ts in both loaders. When we dismiss the cjs one we can have one implementation

Reply:
Why do we need to support both loaders? The ESM loader supports CommonJS. When I say keep the implementation within the ESM loader, I'm not suggesting dropping support for CommonJS; I'm saying you probably don't need to implement this twice.

Reply:
I think it should be possible and it would simplify things considerably; and in general we want to leave the CommonJS loader in peace if at all possible since it’s so easy to break monkey-patchers with any changes there. If it’s not easy then we can address in a follow-up.

Reply:
it's not straightforward and probably there are some thing to figure out with require-esm